src/metabase/actions/execution.clj

```diff
@@ -172,16 +162,29 @@
 (defn execute-action!
   "Execute the given action with the given parameters of shape `{<parameter-id> <value>}."
   [action request-parameters]
-  (let [; add default values for missing parameters
+  (let [; if a value is supplied for a hidden parameter, it should raise an error
+        ; this is to prevent users from supplying a value for a hidden parameter
+        field-settings         (get-in action [:visualization_settings :fields])
+        hidden-param-ids       (->> (vals field-settings)
+                                    (filter :hidden)
+                                    (map :id))
+        destination-param-ids  (set/difference (set (map :id (:parameters action))) (set hidden-param-ids))
+        _ (doseq [[parameter-id _value] request-parameters]
```

Comment:
how about we move this into a helper function? something like `check-no-hidden-parameter-supplied`

Reply:
Done [Extract check-no-extra-parameters](https://github.com/metabase/metabase/pull/30904/commits/878da2977a6faa9ba6d7a2ee5cc68a446ecaf869)